woocommerce/woocommerce-ios
# Order Creation: Display selected variation attributes in product list

Closes: #6031

## Description

When adding a variation to a new order during Order Creation, we now display the variation attributes (instead of the stock status) in the product list on the New Order screen.

## Changes

1. We use the list of variation attributes in several places (in the Products and Orders tabs). The list is generated from the parent product's attributes (to include attributes like "Any Size" where the variation does not have a specified value). To avoid duplicating the logic to create that list, and the variation name, I consolidated that logic in `ProductVariation+Woo` so it can be reused.
2. `ProductRowViewModel` is updated to take an optional list of variation attributes. When the attributes are provided to the view model, they replace the stock status in the product details label. (For example, instead of "In stock • $5.00" the label will be "Blue, Any Size • $5.00") This is optional to allow variations to show their stock status in the variation list before they are added to the order.
3. `NewOrderViewModel` is updated to include the attributes list when creating the product row view models, so the attributes appear in the product list on the new order.

## Testing

1. Build and run the app.
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab, tap the + button, and select "Create order."
4. On the New Order screen, tap the "Add product" button.
5. Select a variable product. Confirm the variations list shows each variation's stock status.
6. Select a variation. On the New Order screen, confirm the variation appears in the Products section with a list of its attributes instead of its stock status.

## Screenshots

Variations in variation list (before adding to order)|Variations in product list on the new order
-|-
![Simulator Screen Shot - iPhone 13 Pro - 2022-02-10 at 18 29 53](https://user-images.githubusercontent.com/8658164/153475424-a2ddac91-54ef-4bf4-8bd1-91f05c1372c1.png)|![Simulator Screen Shot - iPhone 13 Pro - 2022-02-10 at 18 30 23](https://user-images.githubusercontent.com/8658164/153475440-84c4c392-7ac9-4d89-89b6-c91fdbdc107c.png)


## Submitter Checklist

Update release notes:
